maven-core/src/test/java/org/apache/maven/repository/TestRepositoryConnector.java

```diff
@@ -49,10 +49,18 @@ public class TestRepositoryConnector implements RepositoryConnector {
 
     public TestRepositoryConnector(RemoteRepository repository) {
         this.repository = repository;
-        try {
-            basedir = FileUtils.toFile(new URL(repository.getUrl()));
-        } catch (MalformedURLException e) {
-            throw new IllegalStateException(e);
+        String repositoryUrl = repository.getUrl();
+        if (repositoryUrl.contains("${")) {
```

Comment:
This is somewhat ugly, but I haven't really found a better way at this point (without some major changes in how the `TestRepositoryConnector` is being used.

Basically either the `TestRepositoryConnector` would need to resolve the properties in the URL, or the constructor would have to be called with the `repository` which already has these resolved.